### PR TITLE
Add self-update command

### PR DESCRIPTION
**Pull request by kendru**

### CODE REVIEW

File: telemetry/autoinstalled.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+	if err := os.Chmod(tmpfile.Name(), 0700); err != nil {
+		return fmt.Errorf("making script executable: %w", err)
+	}
+
+	out, err := exec.Command(tmpfile.Name()).Output()
+	if err != nil {
+		return fmt.Errorf("running update script: %w", err)
+	}
+	fmt.Printf(string(out))
```
Comment (brandonbloom) [inline telemetry/autoinstalled.go:42]:
```go
cmd.Stdout = os.Stdout
cmd.Stderr = os.Stderr
```

File: core/version.go [hunk 2]
```diff
@@ -2,5 +2,10 @@ package core
 
 import _ "embed"
 
+const (
+	CheckVersionEndpoint = "https://download-page.deref.workers.dev/version.txt"
+	UpdateScriptEndpoint = "https://download-page.deref.workers.dev/install.sh"
```
Comment (brandonbloom) [inline core/version.go:7]: We don't control `workers.dev`, so let's use the redirecting endpoint to insulate ourselves from the implementation service.

**Review by brandonbloom — APPROVED** [review]
LGTM except we should use URLs for which we fully control the domain.

**Comment by kendru** [comment]
Fixes #4

**Pull request merged**